lib/spack/spack/test/bindist.py

```diff
@@ -148,18 +120,21 @@ def default_config(tmpdir_factory, config_directory, monkeypatch):
     timeout = spack.config.get('config:connect_timeout')
     if not timeout:
         spack.config.set('config:connect_timeout', 10, scope='user')
+
     yield spack.config.config
+
+    spack.config.config = old_config
     mutable_dir.remove()
 
 
 @pytest.fixture(scope='function')
 def install_dir_default_layout(tmpdir):
     """Hooks a fake install directory with a default layout"""
-    real_store = spack.store.store
-    real_layout = spack.store.layout
-    spack.store.store = spack.store.Store(str(tmpdir.join('opt')))
-    spack.store.layout = YamlDirectoryLayout(str(tmpdir.join('opt')),
-                                             path_scheme=def_install_path_scheme)  # noqa: E501
+    scheme = '${ARCHITECTURE}/${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}-${HASH}'  # noqa: E501
```

Comment:
Can we use the non-deprecated format strings here?

Reply:
@becker33 Should be done